modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/extras/RankFeatureFieldMapper.java

```diff
@@ -122,6 +155,9 @@ public ValueFetcher valueFetcher(SearchExecutionContext context, String format)
             return new SourceValueFetcher(name(), context) {
                 @Override
                 protected Float parseSourceValue(Object value) {
+                    if (value.equals("")) {
```

Comment:
This line was added because this assertion was failing.
`assertEquals(List.of(2.0f), fetchSourceValue(mapper, ""));` (RankFeatureFieldTypeTests)
This assertion was added to test the null case.  I followed the implementation in ScaledFloatFieldTypeTests but perhaps they were not similar. Perhaps this line is not needed and instead, the nullValue should be passed to SourceValueFetcher constructor so that the assertion succeeds with null instead of "".

Reply:
@javanna Thanks for checking. Indeed we would never get an empty string, as unlike numeric fields we don't allow to index empty strings for `rank_feature` fields. Addressed in  https://github.com/elastic/elasticsearch/pull/96392